docs/aws/audit/kubernetesmonitoring/rules/ecs_awsvpc_networking_enabled_remediation.mdx

```diff
@@ -1,6 +1,224 @@
+
+   # Replace 'your-cluster-name' with your actual ECS cluster name
+   enforce_awsvpc_network_mode('your-cluster-name')
+   ```
+
+4. **Run the Script**:
+   Execute the script to enforce the `awsvpc` network mode for all ECS task definitions.
+
+   ```bash
+   python enforce_awsvpc_network_mode.py
+   ```
+
+This script will iterate through all ECS task definitions, check their network mode, and update them to use `awsvpc` if they are not already configured to do so.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the ECS (Elastic Container Service) section.
+
+2. In the ECS section, select the cluster where your tasks are running.
+
+3. Once you've selected the cluster, navigate to the "Tasks" tab. Here, you will see a list of all the tasks running in the selected cluster.
+
+4. Click on the task name to view its details. In the "Task Definition" section, look for the "Network Mode". If it's not set to "awsvpc", then it's a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all the ECS task definitions in your account using the following command:
+
+   ```
+   aws ecs list-task-definitions --region your-region-name
+   ```
+
+   Replace "your-region-name" with the name of the AWS region where your ECS tasks are running.
+
+3. For each task definition ARN returned by the above command, you can describe the task definition to get its network mode using the following command:
+
+   ```
+   aws ecs describe-task-definition --task-definition task-definition-arn --region your-region-name
+   ```
+
+   Replace "task-definition-arn" with the ARN of the task definition you want to check, and "your-region-name" with the name of the AWS region where your ECS tasks are running.
+
+4. The above command will return a JSON output. You can check the "networkMode" field in the output. If the "networkMode" is not set to "awsvpc", then the ECS task does not have its network mode set to AWSVPC.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if ECS Tasks have Network Mode set to AWSVPC in Kubernetes using Python scripts, you can use the Boto3 library, which allows you to directly interact with AWS services, including ECS. Here are the steps:
+
+1. **Import the necessary libraries and establish a session with AWS:**
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>